numpy/lib/tests/test_function_base.py

```diff
@@ -1225,6 +1225,10 @@ def test_no_trim(self):
         assert_array_equal(arr, res)
 
 
+    def test_list_to_list(self):
+        res = trim_zeros(self.a.tolist())
+        assert isinstance(res, list)
```

Comment:
I assume this test does not pass on master?

Reply:
It does as `self.a` is just a normal 1D array of integers.

However, there is currently no test which checks the return type so I feel this is a nice addition.

Reply:
I'm trying to work out whether this is a case of @mattip generally wishing to reseal the can of worms, or if there was a specific worm that we knew was an incompatibility.

Reply:
Mainly seal the can of worms.  I am a bit dubious if the new implementation actually achieves what it promises, personally.  (Depending on the use-case, you may be stripping only one value from a very large array, making the old code much faster. OTOH, maybe that use-case doesn't actually matter since in almost all cases the work done on the result will be much slower anyway.)

If you are fairly certain the updated version seals it well, I think we can go with that as well.  I had never looked very carefully, but it seemed a bit hard to be sure previously, and `trim_zeros` seemed a bit niche to worry about much...

Reply:
The PR reverts the code to its state before the first PR gh-16911. I added the passing test to cover a use case that previously was not tested.